cmake/OpenCVDetectCXXCompiler.cmake

```diff
@@ -194,54 +194,7 @@ if(CMAKE_VERSION VERSION_LESS "3.1")
   endforeach()
 endif()
 
-if(NOT OPENCV_SKIP_CMAKE_CXX_STANDARD)
-  ocv_update(CMAKE_CXX_STANDARD 11)
-  ocv_update(CMAKE_CXX_STANDARD_REQUIRED TRUE)
-  ocv_update(CMAKE_CXX_EXTENSIONS OFF) # use -std=c++11 instead of -std=gnu++11
-  if(CMAKE_CXX11_COMPILE_FEATURES)
-    set(HAVE_CXX11 ON)
-  endif()
-endif()
-if(NOT HAVE_CXX11)
-  ocv_check_compiler_flag(CXX "" HAVE_CXX11 "${OpenCV_SOURCE_DIR}/cmake/checks/cxx11.cpp")
-  if(NOT HAVE_CXX11)
-    ocv_check_compiler_flag(CXX "-std=c++11" HAVE_STD_CXX11 "${OpenCV_SOURCE_DIR}/cmake/checks/cxx11.cpp")
-    if(HAVE_STD_CXX11)
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-      set(HAVE_CXX11 ON)
-    endif()
-  endif()
-endif()
-
-if(NOT HAVE_CXX11)
-  message(FATAL_ERROR "OpenCV 4.x requires C++11")
-endif()
-
-set(__OPENCV_ENABLE_ATOMIC_LONG_LONG OFF)
-if(HAVE_CXX11 AND (X86 OR X86_64))
-  set(__OPENCV_ENABLE_ATOMIC_LONG_LONG ON)
-endif()
-option(OPENCV_ENABLE_ATOMIC_LONG_LONG "Enable C++ compiler support for atomic<long long>" ${__OPENCV_ENABLE_ATOMIC_LONG_LONG})
-
-if((HAVE_CXX11 AND OPENCV_ENABLE_ATOMIC_LONG_LONG
-        AND NOT MSVC
-        AND NOT (X86 OR X86_64)
-    AND NOT OPENCV_SKIP_LIBATOMIC_COMPILER_CHECK)
-    OR OPENCV_FORCE_LIBATOMIC_COMPILER_CHECK
-)
-  ocv_check_compiler_flag(CXX "" HAVE_CXX_ATOMICS_WITHOUT_LIB "${OpenCV_SOURCE_DIR}/cmake/checks/atomic_check.cpp")
-  if(NOT HAVE_CXX_ATOMICS_WITHOUT_LIB)
-    list(APPEND CMAKE_REQUIRED_LIBRARIES atomic)
```

Comment:
Until you have reproducer for corresponding issue please do not change checks. Not all configuration cases are tested in these 10 public builders.

Use `git blame` (GitHub was Web UI for that) to find corresponding commits, its notes, PRs and linked issues.